tests/beaker_tests/snmp_user/snmp_userlib.rb

```diff
@@ -0,0 +1,99 @@
+# ---------------------
+# snmp_userlib.rb
+#
+# This is the utility library for the Snmp User provider Beaker test cases
+# that contains the common methods used across the Snmp User testsuite's
+# cases. The library is implemented as a module with related methods and
+# constants defined inside it for use as a namespace. All of the methods are
+# defined as module methods.
+#
+# Every Beaker Snmp User test case that runs an instance of Beaker::TestCase
+# requires RadiusSettingLib module.
+#
+# The module has a single set of methods:
+# A. Methods to create manifests for snmp_user Puppet provider test cases.
+###############################################################################
+
+# Require UtilityLib.rb path.
+require File.expand_path('../../lib/utilitylib.rb', __FILE__)
+
+# A library to assist testing snmp_user resource
+module SnmpUserLib
+  # Group of Constants used in negative tests for snmp_user provider.
+  ENSURE_NEGATIVE = 'unknown'
+
+  # A. Methods to create manifests for snmp_user Puppet provider test cases.
+
+  # Method to create a manifest for snmp_user resource attribute 'ensure'
+  # where 'ensure' is set to present.
+  # @param none [None] No input parameters exist.
+  # @result none [None] Returns no object.
+  def self.create_snmp_user_manifest_present
+    manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+node default {
+  snmp_user { 'test_snmp_user':
+    ensure          => present,
+    roles           => ['network-operator'],
+    auth            => 'md5',
+    password        => '0x7e5030ffd26d7e1b366a9041e9c63c94',
+    privacy         => 'aes128',
+    private_key     => '0xcc012f26b3384d4b3da979bff48b4ffe',
+    localized_key   => true,
+  }
+}
+EOF"
+    manifest_str
+  end
+
+  # Method to create a manifest for snmp_user resource attribute 'ensure'
+  # where 'ensure' is set to present, and a few changes made from above.
+  # @param none [None] No input parameters exist.
+  # @result none [None] Returns no object.
+  def self.create_snmp_user_manifest_present_change
+    manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+node default {
+  snmp_user { 'test_snmp_user':
+    ensure          => present,
+    roles           => ['network-operator'],
+    auth            => 'sha',
+    password        => '0x7e5030ffd26d7e1b366a9041e9c63c94',
+    privacy         => 'des',
+    private_key     => '0xcc012f26b3384d4b3da979bff48b4ffe',
+    localized_key   => true,
+  }
+}
+EOF"
+    manifest_str
+  end
+
+  # Method to create a manifest for snmp_user resource attribute 'ensure'
+  # where 'ensure' is set to absent.
+  # @param none [None] No input parameters exist.
+  # @result none [None] Returns no object.
+  def self.create_snmp_user_manifest_absent
+    manifest_str = "cat <<EOF >#{UtilityLib::PUPPETMASTER_MANIFESTPATH}
+node default {
+    snmp_user { 'test_snmp_user':
+     ensure          => absent,
+  }
+}
+EOF"
+    manifest_str
+  end
+end
```

Comment:
@hunsbea To the best of my knowledge, we haven't done negative testing in any of our other netdev providers that we have implemented (and have been merged). It's definitely a nice to have, but from a deliverable standpoint we've been focusing on CRUD tests, in line with the "acceptance" testing done in other Puppet modules.